### DISCUSSION TITLE
Solax - Drazice inverter

### DISCUSSION THREAD

**Discussion by tomasvida** [wills106/homeassistant-solax-modbus#394] [Q&A]
Hello, please help me:
Hi, I have a Drazice (Czech Republic) inverter, which should only be a Solax with a different brand print.
It says on the sticker that it is a model
In.Compact 10.0k
SN MC210TI2......
I can see and control inverter via the website https://www.solaxcloud.com/
where the type shows me as X3-MIC/PRO-G2
Inverter has an adapter for connecting to LAN and an RS485 interface to which a smart meter is connected. I connected the RS485/ETH waveshare converter to the smart meter, configured it according to the instructions, but I am not able to connect to the inverter.
Do I have any chance of connecting this stridac to this integration?

Thanks for help
Tomas

![image](https://user-images.githubusercontent.com/97637008/235674610-1f5bcbc5-6afc-499d-8bd2-91083063ddf9.png)

![image](https://user-images.githubusercontent.com/97637008/235674709-bf372523-92de-45a1-8206-ca4fff57b8e7.png)

![image](https://user-images.githubusercontent.com/97637008/235675010-4ace2352-aa31-4b31-8d7d-44cf1157c5f8.png)

**Comment by wills106** [comment]
Do you get an error in your log that MC210 is an unknown Inverter?
Or do you get an error it can't detect an Inverter at all?

If it can't detect anything at all, what baud rate have you set? It should be 9600.

**Reply by tomasvida** [reply]
My HA Core log:
Home Assistant Core
unrecognized inverter type - serial number : unknown
21:00:22 – (CHYBA) SolaX Inverter Modbus (vlastní integrace) - zpráva se poprvé objevila v 07:21:10 a zobrazuje se 2 krát
SolaX: cannot find serial number, even not for MIC
21:00:22 – (CHYBA) SolaX Inverter Modbus (vlastní integrace) - zpráva se poprvé objevila v 07:21:10 a zobrazuje se 2 krát
SolaX: reading serial number from address 0x300 failed; other address may succeed
21:00:22 – (VAROVÁNÍ) SolaX Inverter Modbus (vlastní integrace) - zpráva se poprvé objevila v 07:21:10 a zobrazuje se 2 krát
SolaX: reading serial number from address 0x0 failed; other address may succeed
21:00:17 – (VAROVÁNÍ) SolaX Inverter Modbus (vlastní integrace) - zpráva se poprvé objevila v 07:21:05 a zobrazuje se 2 krát

I think I have Waveshare set up correctly, I followed the source on the Internet. I'm just not sure about the Destination IP/DNS entry... what kind of address is this? My converter has the address 192.168.100.15, which can eventually be seen in the header of the browser, but I don't understand this.
The address of modbuss 2 is correct, I set this first in the menu on the inverter. I don't know what address is set on the smart meter, if there is one at all, I don't know how to verify it, but the inverter communicates with the smart meter, so the connection is functional. The RS485/Ethernet converter is connected in parallel to the smart meter with an approx. 10 cm long cable. I don't have a termination resistor at the converter, does it have to be there?

![image](https://user-images.githubusercontent.com/97637008/235761323-cdda6b10-024c-4227-bae2-09f216acb018.png)

![image](https://user-images.githubusercontent.com/97637008/235762103-3dc25172-f2be-4c26-9303-2d1507e1d646.png)

![image](https://user-images.githubusercontent.com/97637008/235761586-8fa85839-9caa-408b-abdb-3fdd59a78222.png)

Tomas

**Reply by wills106** [reply]
Have you tried swapping the wires around on the Waveshare? Also do you need to actually turn Modbus on in the Inverter menu on these Inverters?

**Comment by tomasvida** [comment]
Hello,
swapping wires without effect, I'm 100% sure that I have it connected correctly, according to the smart meter, which is right next to the Waveshare converter.
And this is my settings in solarpower.
![image](https://github.com/wills106/homeassistant-solax-modbus/assets/97637008/13815757-c79d-43b0-a225-1911e403ddfc)

Tomas

**Comment by tomasvida** [comment]
Hello, 
i have new error with unrecognized inverter type:

Tato chyba pochází z vlastní integrace.

Logger: custom_components.solax_modbus.plugin_solax
Source: custom_components/solax_modbus/plugin_solax.py:4746
Integration: SolaX Inverter Modbus (documentation, issues)
First occurred: 09:59:36 (2 occurrences)
Last logged: 10:04:15

unrecognized inverter type - serial number : MC210TI2136015
unrecognized inverter type - serial number : unknown

**Reply by wills106** [reply]
You have working communications now. What did you have to change?

I have just released [2023.06.1](https://github.com/wills106/homeassistant-solax-modbus/releases/tag/2023.06.1) which includes your Serial.

**Comment by tomasvida** [comment]
I had to turn off the communication between the inverter and the chint electricity meter (advancedSetting / ExportControl/Mode_Select - DISABLE/. If the communication is on, I cannot read anything from the inverter. Even so, the communication is very unreliable and breaks.
So I had to set the pooling freq to 5 sec. If it is higher, the data cannot be loaded.
Is there a reason why the integration only shows me 29 basic entities?

The supplier (Drazice) sent me a manual for the registers, doesn't it help you?

![image](https://github.com/wills106/homeassistant-solax-modbus/assets/97637008/da2774cc-96a0-40fc-853b-da953f90db7f)
[Hybrid X1&X3-G4 ModbusRTU V3.14-English - public version (1).pdf](https://github.com/wills106/homeassistant-solax-modbus/files/11704208/Hybrid.X1.X3-G4.ModbusRTU.V3.14-English.-.public.version.1.pdf)

**Reply by wills106** [reply]
That's the Modbus registers for the Gen4 Hybrid / AC Storage.
The X3 MIC / PRO has a vey limited set of registers compared to the Hybrids and they are in a different location. There aren't any other registers of interest on your Inverter unfortunately.

**Comment by tomasvida** [comment]
Hi, please, can you add my next inverter to integration
MC210TI2136005

**Reply by wills106** [reply]
Any Inverter that starts MC210T is supported.
The Integration only needs to know the first 6 digits to determine what Model it is.

Edit: How it works
https://github.com/wills106/homeassistant-solax-modbus/blob/202b7cf199b7a5787f51b56b70cd1fa58a4f45f1/custom_components/solax_modbus/plugin_solax.py#L5218-L5224
